backend/utilities/helpers/AzureBlobStorageHelper.py

```diff
@@ -80,8 +114,14 @@ def upsert_blob_metadata(self, file_name, metadata):
 
     def get_container_sas(self):
         # Generate a SAS URL to the container and return it
-        return "?" + generate_container_sas(account_name= self.account_name, container_name= self.container_name,account_key=self.account_key,  permission="r", expiry=datetime.utcnow() + timedelta(hours=1))
+        if self.use_rbac:
+            return "?" + generate_container_sas(account_name= self.account_name, container_name= self.container_name, user_delegation_key=self.user_delegation_key,  permission="r", expiry=datetime.utcnow() + timedelta(hours=1))
+        else:
+            return "?" + generate_container_sas(account_name= self.account_name, container_name= self.container_name,account_key=self.account_key,  permission="r", expiry=datetime.utcnow() + timedelta(hours=1))
```

Comment:
I feel like we need to abstract this "auth type" logic into a class and then have the class manage what is returned.

I'm seeing too much duplicated forking code.  Can you come up with a better design?

Reply:
We have updated the code and no longer repeatedly determine whether to use `rbac`.